src/lib/bootloader/bls_sections.rb

```diff
@@ -16,27 +16,35 @@ class BlsSections
     # or an empty array
     attr_reader :all
 
-    # @return [String] title of default boot section.
-    attr_reader :default
-
     def initialize
       @all = []
       @default = ""
     end
 
+    # @return [String] title of default boot section.
+    def default
+      return unless @data
+
+      entry = @data.select { |d| d["id"] == @default }
+      if entry.empty?
+        ""
+      else
+        entry.first["title"]
```

Comment:
this looks strange as you basically search for first entry, so it should look like
```ruby
entry = @data.find { |d| d["id"] == @default }
entry ? entry["title"] : ""
```